packages/synapse-react-client/src/components/SynapseForm/SynapseForm.tsx

```diff
@@ -827,10 +828,30 @@ export default class SynapseForm extends React.Component<
   render() {
     return (
       <div className="outter-wrap">
-        <Prompt
-          when={this.state.hasUnsavedChanges}
-          message={this.unsavedDataWarning}
-        />
+        <Blocker
```

Comment:
Created a simple `Blocker` component that wraps `useBlocker` so we can easily use it in this class component.